iwyu_globals.cc

```diff
@@ -328,15 +327,16 @@ void InitGlobals(clang::SourceManager* sm,
   function_calls_full_use_cache = new FullUseCache;
   class_members_full_use_cache = new FullUseCache;
 
-  for (Each<HeaderSearchPath> it(&search_paths); !it.AtEnd(); ++it) {
-    const char* path_type_name
-        = (it->path_type == HeaderSearchPath::kSystemPath ? "system" : "user");
-    VERRS(6) << "Search path: " << it->path << " (" << path_type_name << ")\n";
+  for (const HeaderSearchPath& entry : search_paths) {
+    const char* path_type_name =
+        (entry.path_type == HeaderSearchPath::kSystemPath ? "system" : "user");
+    VERRS(6) << "Search path: " << entry.path << " (" << path_type_name
+             << ")\n";
```

Comment:
I agree that the formatting of this line is a little weird, but I've tried to just defer to clang-format's judgment, and this is it. The more aligned we are with clang-format, the easier it is moving forward, in my opinion.

I could try and reconstruct the logic to not use the ternary operator, but I don't see that it will be clearer.